src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -120,8 +120,17 @@ size_t length = sizeof(cpu_load_info);
     if (sysctl(mib, miblen, &cpu_load_info, &length, NULL, 0) == -1) {
       return OS_ERR;
     }
+#elif defined(__FreeBSD__)
+    size_t alllength = length * nProcs * sizeof(long);
+    long *allcpus = NEW_C_HEAP_ARRAY(long, alllength, mtInternal);
+
+    if (sysctlbyname("kern.cp_times", allcpus, &alllength, NULL, 0) == -1) {
+      return OS_ERR;
```

Comment:
`sysctlbyname` third parameter is in bytes but `alllength` is not correctly set. length is already in bytes so multiplying to by sizeof(long) is not correct. `size_t alllength = length * nProcs;` is correct for this value.

Also the memory allocated for allcpus is leaked. Please use `FREE_C_HEAP_ARRAY` to deallocate.